### ISSUE TITLE
Maintenance of ChangeLog and NEWS files

### ISSUE THREAD

**Issue by Quintus** [mruby/mruby#3190]
Hi,

it appears the `ChangeLog` and `NEWS` file are unmaintained (they contain no information). I would appreciate it if the `NEWS` file would duplicate the information from the release announcements to have them readable in the source tree itself, but if there is no intention to maintain either of these files, they should be removed, because it looks confusing otherwise.

Greetings
Marvin

**Comment by matz** [comment]
You're right. I will remove ChangeLog. Keeping NEWS seems a good idea, if we can maintain. But we clearly haven't done it well, should we remove it too?

**Comment by bovi** [comment]
I'm all for removing it or creating a script to generate it out of the git logs for a release candidate.

**Comment by Quintus** [comment]
On Sat, Aug 06, 2016 at 04:35:39AM -0700, Daniel Bovensiepen wrote:
> I'm all for removing it or creating a script to generate it out of the
> git logs for a release candidate.

No, please not. The Git log is not as informative to someone who does
not actively follow the development. The NEWS file should contain the
user-visible changes, which you cannot automatically generate from the
Git log.

Greetings
Marvin

**Comment by bovi** [comment]
> The NEWS file should contain the user-visible changes, which you cannot automatically generate from the Git log.

I agree fully but most projects I see can't maintain something like that in the long run. So better than nothing or even better then something wrong, I think this would be an option which needs low efforts. And in the end nothing speaks against compressing the git-log output to something more understandable to the general public.

**Comment by zzak** [comment]
I think if you ask people submitting features to add a line to NEWS its much easier to maintain, and don't worry about trying to add every bugfix to the file.

**Comment by Quintus** [comment]
On Sun, Aug 07, 2016 at 05:51:04AM -0700, Daniel Bovensiepen wrote:
> > The NEWS file should contain the user-visible changes, which you
> > cannot automatically generate from the Git log.
>
> I agree fully but most projects I see can't maintain something like
> that in the long run.

I suggested in the OP that the file should contain copies of the release
announcements. Since these will be written anyway, I don't see much
extra work than copy&paste of the release announcement markup into the
NEWS file. For example, for 1.2.0 it could simply contain the same text
as the 1.2.0 release announcement in the "What's new" section[1]. If the
info is available anyway, why not add it to the source tree?

Greetings
Marvin

[1]: http://mruby.org/releases/2015/11/18/mruby-1.2.0-released.html

**Comment by matz** [comment]
Closing — `ChangeLog` / `NEWS` maintenance question. NEWS.md exists, and per-release notes live under `doc/mruby*.md` (e.g., doc/mruby4.0.md). git log provides authoritative history.
